src/icon-button.tsx

```diff
@@ -144,26 +154,200 @@ const RawIconButton: React.FC<Props> = (props) => {
+                        if (showSpinner !== shouldRenderSpinner) {
+                            setShouldRenderSpinner(showSpinner);
+                        }
+                    }}
+                >
+                    {shouldRenderSpinner && (
+                        <Spinner size={styles.iconSize[buttonSize]} color="currentColor" delay="0s" />
+                    )}
+                </div>
+            </div>
+        );
+
+        if (props.href) {
+            return (
+                <BaseTouchable {...commonProps} href={props.href} newTab={props.newTab}>
+                    {content}
+                </BaseTouchable>
+            );
+        }
+        if (props.to) {
+            return (
+                <BaseTouchable
+                    {...commonProps}
+                    to={props.to}
+                    fullPageOnWebView={props.fullPageOnWebView}
+                    replace={props.replace}
+                >
+                    {content}
+                </BaseTouchable>
+            );
+        }
+
+        if (props.onPress) {
+            return (
+                <BaseTouchable
+                    {...commonProps}
+                    onPress={(e) => {
+                        const result = props.onPress(e);
+                        if (result) {
+                            setIsOnPressPromiseResolving(true);
+                            result.finally(() => setIsOnPressPromiseResolving(false));
+                        }
+                    }}
+                >
+                    {content}
+                </BaseTouchable>
+            );
+        }
+
+        return (
+            <BaseTouchable {...commonProps} maybe>
+                {content}
+            </BaseTouchable>
+        );
+    }
+);
+
+type IconButtonProps = ExclusifyUnion<OldProps | NewProps>;
+
+export const InternalIconButton = React.forwardRef<
+    TouchableElement,
+    IconButtonProps & {isOverMedia?: boolean}
+>((props, ref) => {
+    if (props.Icon) {
+        return <RawIconButton ref={ref} {...props} />;
+    }
 
-const IconButton = (props: Props): JSX.Element => {
     const {icon, backgroundColor = 'transparent', iconSize, size = ICON_SIZE_1} = props;
     return (
-        <RawIconButton
+        <RawOldIconButton
+            ref={ref}
             {...props}
-            className={classnames(styles.base, props.className)}
+            className={classNames(styles.oldIconButtonBase, props.className)}
             style={{...getButtonStyle(icon, size, backgroundColor, iconSize, props.disabled), ...props.style}}
         />
     );
-};
+});
+
+export const IconButton = React.forwardRef<TouchableElement, IconButtonProps>((props, ref) => {
+    return <InternalIconButton ref={ref} {...props} />;
+});
 
 // Used internally by Mistica's components to avoid styles collisions
-export const BaseIconButton = (props: Props): JSX.Element => {
+
+export const BaseIconButton = (props: IconButtonProps): JSX.Element => {
```

Comment:
This internal component is only used in some input fields. I believe that with this new version of the IconButton, it may not be really necessary anymore.

Once we update all the Mistica components with the new IconButton, I will remove it if it's possible.